airflow/models.py

```diff
@@ -346,10 +346,10 @@ def kill_zombies(self, session=None):
 
                     # now set non db backed vars on ti
                     ti.task = task
-                    ti.test_mode = False
+                    ti.test_mode = configuration.getboolean('core', 'unit_test_mode')
```

Comment:
test_mode is set per instance and not by the configuration, eg. by run. I'm not sure how "test_mode" behaves in context of killing a zombie. Is it really required? If it is it needs to be derived differently